### PR TITLE
Add test cases to test sudoers file

### PR DESCRIPTION
**Pull request by qiluo-msft**
### Description of PR

New test cases to make sure readonly user could run allowed commands and cannot run privileged  commands.

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### CODE REVIEW

File: tests/tacacs/test_ro_user.py [hunk 1]
```diff
@@ -20,6 +20,36 @@ def test_ro_user(localhost, duthosts, rand_one_dut_hostname, creds, test_tacacs)
...
+
+    # Run as readonly use the command `sudo docker ps`
+    dutip = duthost.host.options['inventory_manager'].get_host(duthost.hostname).vars['ansible_host']
+    res = localhost.shell("sshpass -p {} ssh "\
+                          "-o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null "\
+                          "{}@{} sudo docker ps".format(
+            creds['tacacs_ro_user_passwd'], creds['tacacs_ro_user'], dutip))
+
+    # Verify that the command is allowed
+    stdout = res['stdout_lines']
```
Comment (lguohan) [inline tests/tacacs/test_ro_user.py:34]: can you generalize the check. just check the return value, if it non-zero or non-zero.

I'd like you to cover all command defined in the sudoer file in this pr as well.

File: tests/tacacs/test_ro_user.py [hunk 2]
```diff
@@ -20,6 +20,36 @@ def test_ro_user(localhost, duthosts, rand_one_dut_hostname, creds, test_tacacs)
...
+    assert len(stdout) >= 1
+    assert stdout[0].startwith('CONTAINER ID')
```
Comment (lguohan) [inline tests/tacacs/test_ro_user.py:36]: check the return value, do not check the content.

File: tests/tacacs/test_ro_user.py [hunk 3]
```diff
@@ -20,6 +20,36 @@ def test_ro_user(localhost, duthosts, rand_one_dut_hostname, creds, test_tacacs)
...
+    dutip = duthost.host.options['inventory_manager'].get_host(duthost.hostname).vars['ansible_host']
+    res = localhost.shell("sshpass -p {} ssh "\
+                          "-o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null "\
+                          "{}@{} sudo docker ps -a".format(
+            creds['tacacs_ro_user_passwd'], creds['tacacs_ro_user'], dutip))
+
+    # Verify that the command is allowed
+    stdout = res['stdout_lines']
+    assert len(stdout) >= 1
+    assert stdout[0].startwith('CONTAINER ID')
```
Comment (lguohan) [inline tests/tacacs/test_ro_user.py:51]: check return value, do not check the content.

**Comment by qiluo-msft** [comment]
Retest vsimage please

**Comment by lguohan** [comment]
can you add the test to kvmtesh t0 suite so that provides check will do the check.

**Comment by qiluo-msft** [comment]
If I read it correctly, the test cases are already there https://github.com/qiluo-msft/sonic-mgmt/blob/9aabf66662cf165d42bbcbc2f5a888f500da4c73/tests/kvmtest.sh#L142

**Comment by lguohan** [comment]
I do not think so. can you check the test results to confirm?

File: tests/tacacs/test_ro_user.py (hunk 1 above)
Comment (lguohan) [reply tests/tacacs/test_ro_user.py]: can you address  this?

**Review by lguohan — CHANGES_REQUESTED** [review]
please test all ro only cmd defined in the summer file

**Comment by qiluo-msft** [comment]
I see the test result in https://dev.azure.com/sonicswitch/be1b070f-be15-4154-aade-b1d3bfb17054/_apis/build/builds/2396/logs/20
```
2021-01-31T08:35:14.3026790Z tacacs/test_ro_user.py::test_ro_user PASSED                              [ 25%]
2021-01-31T08:35:15.6780982Z tacacs/test_ro_user.py::test_ro_user_docker_ps PASSED                    [ 50%]
2021-01-31T08:35:17.0877996Z tacacs/test_ro_user.py::test_ro_user_docker_ps_all PASSED                [ 75%]
2021-01-31T08:35:35.9374924Z tacacs/test_ro_user.py::test_ro_user_ipv6 PASSED                         [100%]
```

**Comment by lguohan** [comment]
got it

File: tests/tacacs/test_ro_user.py [hunk 4]
```diff
@@ -20,6 +20,32 @@ def test_ro_user(localhost, duthosts, rand_one_dut_hostname, creds, test_tacacs)
         if fds[0] == "test":
             assert fds[4] == "remote_user"
 
+def test_ro_user_docker_ps(localhost, duthosts, rand_one_dut_hostname, creds, test_tacacs):
```
Comment (lguohan) [inline tests/tacacs/test_ro_user.py:23]: suggest you to combine test_ro_user_docker_ps and test_ro_user_docker_ps_all as a function test_ro_user_sudo_cmds to check all commands that matches the sudoer file.

**Review by lguohan — APPROVED** [review]

**Comment by lguohan** [comment]
can you please change the pr title and content accordingly?

**Pull request merged**